### PR TITLE
Fix errors when passing and returning NULL

### PR DESCRIPTION
**Pull request by xuebinsu**
This patch contains bug fixes regarding passing and returning NULL between PL/Container and R Client. Note that there are more lines changed than you might expect since the formatter would format my code on save. Some other changes are introduced for debugging. The key part is the big if statement testing NA values. 

Please review and merge it. Thanks.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@xuebinsu Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=greenplum-db/rclient&pullRequestId=45)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=greenplum-db/rclient&pullRequestId=45) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@xuebinsu Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=greenplum-db/rclient&pullRequestId=45)!

File: bin/rclient.sh [hunk 1]
```diff
@@ -6,5 +6,5 @@
 # ------------------------------------------------------------------------------
 cd /clientdir
 
-./rclient
+./rclient &> rclient.log
```
Comment (HaozhouWang) [inline bin/rclient.sh]: Output to *.log should be removed

File: client.c [hunk 2]
```diff
@@ -4,47 +4,44 @@
...
+#include "common/comm_utils.h"
 #include "rcall.h"
 
 int main(int argc UNUSED, char **argv UNUSED) {
-	int sock;
-	plcConn *conn;
-	int status;
+  int sock;
+  plcConn *conn;
+  int status;
```
Comment (HaozhouWang) [inline client.c]: Ident error, please use tab for ident.

File: rcall.c [hunk 3]
```diff
@@ -676,28 +676,38 @@ static int process_call_results(plcConn *conn, SEXP retval, plcRFunction *r_func
 		plc_r_copy_type(&res->types[0], &r_func->res);
 		res->names[0] = strdup(r_func->res.argName);
 
+		plc_elog(DEBUG1, "retval is null = %d", retval == R_NilValue);
 		if (retval == R_NilValue) {
 			res->data[0][0].isnull = 1;
 			res->data[0][0].value = NULL;
-
+			plc_elog(DEBUG1, "returing R_NilValue");
 		} else {
```
Comment (HaozhouWang) [inline rcall.c]: this debug point could be removed

File: client.c [hunk 4]
```diff
@@ -4,47 +4,44 @@
...
-	set_signal_handlers();
+  set_signal_handlers();
 
-	/* do not overwrite, if the CLIENT_NAME has already set */
-	setenv("CLIENT_LANGUAGE", "rclient", 0);
+  /* do not overwrite, if the CLIENT_NAME has already set */
+  setenv("CLIENT_LANGUAGE", "rclient", 0);
 
-	client_log_level = WARNING;
+  client_log_level = DEBUG1;
```
Comment (HaozhouWang) [inline client.c]: change to WARNING as default

**Pull request closed without merging**